pombola/south_africa/management/commands/south_africa_mirror_mapit_wards.py

```diff
@@ -0,0 +1,64 @@
+from contextlib import closing
+import os
+from tempfile import NamedTemporaryFile
+import time
+from urlparse import urljoin
+
+import requests
+
+from django.contrib.gis.gdal import DataSource
+from django.core.management.base import BaseCommand
+from django.db import transaction
+
+from mapit.models import Area, Country, Geometry, Generation, Type
+from mapit.management.command_utils import save_polygons
+
+SLEEP_BETWEEN_REQUESTS = 1
+
+class Command(BaseCommand):
+
+    def handle(self, *args, **options):
+        remote_mapit_url = 'http://mapit.code4sa.org/'
+        all_wards_url = urljoin(remote_mapit_url, '/areas/WD')
+        r = requests.get(all_wards_url)
+        time.sleep(SLEEP_BETWEEN_REQUESTS)
+        with transaction.atomic():
+            # Remove existing wards from MapIt:
+            Geometry.objects.filter(area__type__code='WD').delete()
+            Area.objects.filter(type__code='WD').delete()
+            # Get the Country object for South Africa:
```

Comment:
It's in a `transaction.atomic`, so you should get a complete update or nothing at all.

On a related matter, though, I'm a bit uneasy about removing everything and recreating, just in case someone tries to create foreign keys to areas of type `WD`, since the cascading delete behaviour will remove the other objects. I don't know if "don't do that then" (since they'll discover pretty quickly) is a good enough reponse to this. (Even if we changed the code to only add / remove differences, there's still the risk that someone pointing foreign keys at areas of type `WD` will have objects deleted unexpectedly if they disappear from the code4sa mapit - at least deleting all and recreating will make it really obvious.)